libs/aws/langchain_aws/chat_models/bedrock_converse.py

```diff
@@ -812,6 +834,24 @@ def _stream(
                 kwargs, excluded_keys={"inputSchema", "properties", "thinking"}
             ),
         )
+
+        # Check if toolConfig is missing but tool blocks are present
+        has_tool_blocks = _has_tool_use_or_result_blocks(bedrock_messages)
+        has_tool_config = params.get("toolConfig") is not None
+
+        # Handle conversion
+        if has_tool_blocks and not has_tool_config:
+            logger.warning(
+                "Tool messages (toolUse/toolResult) detected without toolConfig. Converting tool blocks to text format to avoid ValidationException."
+            )
+            warnings.warn(
+                "Tool messages were passed without toolConfig, converting to text format",
+                RuntimeWarning,
+            )
+
+            bedrock_messages = _convert_tool_blocks_to_text(bedrock_messages)
+            logger.debug(f"converted input messages: {bedrock_messages}")
```

Comment:
^Same exit early optimization change for `stream`:
```suggestion
        # Check for tool blocks without toolConfig and handle conversion
        if params.get("toolConfig") is None and _has_tool_use_or_result_blocks(bedrock_messages):
            logger.warning(
                "Tool messages (toolUse/toolResult) detected without toolConfig. "
                "Converting tool blocks to text format to avoid ValidationException."
            )
            warnings.warn(
                "Tool messages were passed without toolConfig, converting to text format",
                RuntimeWarning,
            )

            bedrock_messages = _convert_tool_blocks_to_text(bedrock_messages)
            logger.debug(f"converted input messages: {bedrock_messages}")
```